### ISSUE TITLE
Cannot find postcode from postal_code relation

### ISSUE THREAD

**Issue by tomhughes** [tomhughes/trac-tickets#5140]
**[Submitted to the original trac issue database at 2.20pm, Sunday, 16th March 2014]**

The query for the postalcode "04457" (http://nominatim.openstreetmap.org/search.php?postalcode=04457) results in two adresses:
"Mlkau, Zweinaundorf, Leipzig, Sachsen, 04457, Deutschland, European Union" and
"Lincoln, Piscataquis, Maine, 04457, United States of America" but the result in germany actually has the postalcode 04316.
Furthermore querys for streets in this german region deliver wrong postcodes for the addresses. For example the query (http://nominatim.openstreetmap.org/search.php?q=Paul-Kl%C3%B6psch-Stra%C3%9Fe)delivers "Paul-Klpsch-Strae, Holzhausen, Zweinaundorf, Leipzig, Sachsen, 04288, Deutschland, Europische Union" but the postal code should be 04316.

**Comment by tomhughes** [comment]
**Author: lonvia**
**[Added to the original trac issue at 4.20pm, Sunday, 16th March 2014]**

Outdated postcodes are a known issue. Postcodes have not been updated since the last reimport 1.5 years ago.

The postcode display for the roads needs to be fixed in the OSM data. Looks like there is no postal code area with postcode 04316.

**Comment by tomhughes** [comment]
**Author: defonion**
**[Added to the original trac issue at 5.21am, Saturday, 22nd March 2014]**

But there is an relation with the id 1301664 which has the information of the postal code 04316 surrounding this area. Is this not enough? What would be the way to tag it to be recognized by nominatim?

**Comment by tomhughes** [comment]
**Author: lonvia**
**[Added to the original trac issue at 10.51am, Monday, 24th March 2014]**

Thanks for the relation id. A postal_code area should indeed be enough. Nominatim did import the relation, see [http://nominatim.openstreetmap.org/details.php?osmtype=R&osmid=1301664 here], but it can neither be found nor used in addresses. I've reopened the bug.

It looks like the main issue is that the name only appears in the postcode column, it should be copied as a ref into the name column.

**Comment by tomhughes** [comment]
**Author: lonvia**
**[Added to the original trac issue at 11.27pm, Tuesday, 25th March 2014]**

ref in the name column got lost on updates.

Fixed in [changeset:c2417dd/nominatim]

Note 1: I've manually reimported the relation 1301664 to check that the fix works but other relations with the same problem have not been changed. They will automatically  getsorted out once the data is touched in OSM the next time.

Note 2: this does not fix the well-known issue with postcode centroids (e.g. http://nominatim.openstreetmap.org/details.php?place_id=98362166) not being updated. They still may cause glitches.
